src/info/repo.rs

```diff
@@ -16,159 +14,150 @@ use time::OffsetDateTime;
 use time_humanize::HumanTime;
 
 pub struct Repo<'a> {
-    repo: &'a Repository,
-    logs: Vec<Commit<'a>>,
+    git2_repo: &'a Repository,
+    repo: git::Repository,
+    authors: Vec<Author>,
+    total_num_authors: usize,
+    num_commits: usize,
+    time_of_most_recent_commit: git::actor::Time,
+    time_of_first_commit: git::actor::Time,
 }
 
 #[derive(Hash, PartialEq, Eq)]
 pub struct Sig {
-    name: String,
-    email: String,
+    name: git::bstr::BString,
+    email: git::bstr::BString,
 }
 
-impl From<Signature<'_>> for Sig {
-    fn from(sig: Signature) -> Self {
-        let name = String::from_utf8_lossy(sig.name_bytes()).into_owned();
-        let email = String::from_utf8_lossy(sig.email_bytes()).into_owned();
+impl From<git::actor::Signature> for Sig {
+    fn from(git::actor::Signature { name, email, .. }: git::actor::Signature) -> Self {
         Self { name, email }
     }
 }
 
 impl<'a> Repo<'a> {
     pub fn new(
-        repo: &'a Repository,
+        git2_repo: &'a Repository,
         no_merges: bool,
         bot_regex_pattern: &Option<Regex>,
+        number_of_authors_to_display: usize,
     ) -> Result<Self> {
-        let logs = Self::get_logs(repo, no_merges, bot_regex_pattern)?;
-        Ok(Self { repo, logs })
-    }
+        let mut repo = git::open(git2_repo.path())?;
 
-    fn get_logs(
-        repo: &'a Repository,
-        no_merges: bool,
-        bot_regex_pattern: &Option<Regex>,
-    ) -> Result<Vec<Commit<'a>>> {
-        let mut revwalk = repo.revwalk()?;
-        revwalk.push_head()?;
-        let logs: Vec<Commit<'a>> = revwalk
-            .filter_map(|r| match r {
-                Err(_) => None,
-                Ok(r) => repo
-                    .find_commit(r)
-                    .ok()
-                    .filter(|commit| !(no_merges && commit.parents().len() > 1))
-                    .filter(|commit| {
-                        !(bot_regex_pattern.is_some() && is_bot(commit.author(), bot_regex_pattern))
-                    }),
-            })
-            .collect();
+        // assure that objects we just traversed are coming from cache
+        // when we read the commit right after.
+        repo.object_cache_size(128 * 1024);
```

Comment:
Just curious how `128 * 1024` was chosen. `1024` is 1 KiB? And `128 == 1 << 7`...
Is this supposed to be 1/8 MiB?

Reply:
Actually I made it up 😅. However, I ran [this program](https://github.com/Byron/gitoxide/blob/5052a4e532fec63ccf49f6ce54df41707779e70b/git-repository/examples/stats.rs#L23) and used it to find a better value for this cache. The idea is to keep it just as small as it needs to be to keep the most recently queried object in cache. As each newly added object effectively evicts the previous one, one would want that cache to be small to reduce its overhead.

The adjustment is made in a5ab965a24b97cb0fe33e7b68a8e9fdf8af40701 .

Reply:
Awesome, thanks!